UnchainedLauncher.GUI/ViewModels/ServersTab/ServersTabVM.cs

```diff
@@ -0,0 +1,288 @@
+﻿using CommunityToolkit.Mvvm.Input;
+using LanguageExt;
+using LanguageExt.Pipes;
+using log4net;
+using PropertyChanged;
+using System;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.Diagnostics;
+using System.Linq;
+using System.Net;
+using System.Text.RegularExpressions;
+using System.Threading.Tasks;
+using System.Windows;
+using System.Windows.Input;
+using UnchainedLauncher.Core.API;
+using UnchainedLauncher.Core.API.A2S;
+using UnchainedLauncher.Core.API.ServerBrowser;
+using UnchainedLauncher.Core.JsonModels.Metadata.V3;
+using UnchainedLauncher.Core.Services.Mods;
+using UnchainedLauncher.Core.Services.Processes.Chivalry;
+using UnchainedLauncher.Core.Utilities;
+
+namespace UnchainedLauncher.GUI.ViewModels.ServersTab {
+    using static LanguageExt.Prelude;
+    using static Successors;
+    [AddINotifyPropertyChangedInterface]
+    public class ServersTabVM : IDisposable {
+        private static readonly ILog logger = LogManager.GetLogger(nameof(ServersTabVM));
+        public SettingsVM Settings { get; }
+        public readonly IUnchainedChivalry2Launcher Launcher;
+        public Func<IModManager> ModManagerCreator;
+        public IUserDialogueSpawner DialogueSpawner;
+        public FileBackedSettings<IEnumerable<SavedServerTemplate>>? SaveLocation;
+        public ObservableCollection<ServerTemplateVM> ServerTemplates { get; }
+        public ObservableCollection<(ServerTemplateVM template, ServerVM live)> RunningTemplates { get; } = new();
+        private ServerTemplateVM? _SelectedTemplate;
+        public ServerTemplateVM? SelectedTemplate {
+            get => _SelectedTemplate;
+            set {
+                _SelectedTemplate = value;
+                // this is probably too eager. Save should be triggered when the
+                // template changes, but that's tedious to implement. It's fine for now.
+                Save();
+                UpdateVisibility();
+            }
+        }
+        public ServerVM? SelectedLive { get; private set; }
+
+        public Visibility TemplateEditorVisibility { get; private set; }
+        public Visibility LiveServerVisibility { get; private set; }
+
+        public ICommand Add_Template_Command { get; }
+        public ICommand Remove_Template_Command { get; }
+        public ICommand Launch_Server { get; }
+        public ICommand Launch_Headless { get; }
+        public ICommand Shutdown_Server { get; }
+        public ICommand Save_Command { get; }
```

Comment:
I think these should just be `FooCommand` rather than `Foo_Command`. 

Also, I think we should move away from explicitly defining commands. The MVVM toolkit from microsoft (which we're using) can generate them. https://learn.microsoft.com/en-us/dotnet/communitytoolkit/mvvm/generators/overview

<img width="886" alt="image" src="https://github.com/user-attachments/assets/9fc90b39-d28c-46d9-bdcc-cf9ef171fff7" />